app/models/vacancy/publish.rb

```diff
@@ -0,0 +1,10 @@
+class Vacancy::Publish
+  def initialize(vacancy)
+    @vacancy = vacancy
+  end
+
+  def publish
+    @vacancy.save!
+    NotifyVacancyOnSlackWorker.perform_async(@vacancy.id)
```

Comment:
You still can pass the worker and test it too if you want to test that sidekiq really enqueued the job.